source/MaterialXRuntime/Private/Commands/PvtSetAttributeCmd.cpp

```diff
@@ -4,37 +4,53 @@
 //
 
 #include <MaterialXRuntime/Private/Commands/PvtSetAttributeCmd.h>
+#include <MaterialXRuntime/Private/PvtCommand.h>
 
 #include <MaterialXRuntime/RtPrim.h>
+#include <MaterialXRuntime/RtTypeDef.h>
+#include <MaterialXRuntime/RtValue.h>
 
 namespace MaterialX
 {
 
-PvtSetAttributeCmd::PvtSetAttributeCmd(const RtAttribute& attr, const RtValue& value) :
-    _attr(attr),
-    _value(RtValue::clone(attr.getType(), value, attr.getParent()))
+PvtSetAttributeCmd::PvtSetAttributeCmd(const RtObject& obj, const RtToken& name, const RtValue& value)
+    : PvtCommand()
+    , _obj(obj)
+    , _name(name)
+    , _value(value)
+    , _oldValue()
+    , _attrCreated(false)
 {
 }
 
-PvtCommandPtr PvtSetAttributeCmd::create(const RtAttribute& attr, const RtValue& value)
+PvtCommandPtr PvtSetAttributeCmd::create(const RtObject& obj, const RtToken& name, const RtValue& value)
 {
-    return std::make_shared<PvtSetAttributeCmd>(attr, value);
+    return std::make_shared<PvtSetAttributeCmd>(obj, name, value);
 }
 
 void PvtSetAttributeCmd::execute(RtCommandResult& result)
 {
-    if (_attr.isValid())
+    if (_obj.isValid())
     {
         try
         {
             // Send message that the attribute is changing
-            msg().sendSetAttributeMessage(_attr, _value);
+            msg().sendSetAttributeMessage(_obj, _name, _value);
+
+            RtTypedValue* attr = _obj.getAttribute(_name, RtType::STRING);
```

Comment:
Just checking, are these always strings?

Reply:
Yes, for now. This is just a file rename, and name convention change, from the metadata commands you implemented previously. And they only had string support. But we should probably fix this in another change list, so all attribute types can be set by this command.